index.js

```diff
@@ -68,6 +77,13 @@ export const Router = (props) => {
   // https://reactjs.org/docs/hooks-faq.html#how-to-create-expensive-objects-lazily
   const value = ref.current || (ref.current = { v: buildRouter(props) });
 
+  // Use the "latest ref" pattern.
+  // https://epicreact.dev/the-latest-ref-pattern-in-react/
+  useLayoutEffect(() => {
+    value.v.relbase = props.base || "";
+    value.v.parent = props.parent;
+  });
```

Comment:
@jvdsande yes: your approach definitely won't work in concurrent mode with React 18! Because the order of render operations could be unpredictable. Your ref might contain values from a stale and/or cancelled render operation, making it unreliable! This is why side effects must never happen in render. (Lazy initialization is the big exception, because it's not really a side effect.)

Reply:
Ah yes, definitely. Did not think of that :)